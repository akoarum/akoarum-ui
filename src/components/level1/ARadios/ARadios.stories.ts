import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import { withKnobs } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import ARadios from './ARadios.vue'

storiesOf('level1/ARadios', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { ARadios },
    data() {
      return {
        value: '',
        items: [
          { id: 0, label: '項目1', value: 1 },
          { id: 1, label: '項目2', value: 2 },
          { id: 2, label: '項目3', value: 3 }
        ]
      }
    },
    template: `
      <ARadios v-model="value" :items="items" name="radio" />
    `
  }), { info: {} })
  .add('Error', () => ({
    components: { ARadios },
    data() {
      return {
        value: '',
        items: [
          { id: 0, label: '項目1', value: 1 },
          { id: 1, label: '項目2', value: 2 },
          { id: 2, label: '項目3', value: 3 }
        ]
      }
    },
    template: `
      <ARadios v-model="value" :items="items" name="radio" aria-invalid="true" />
    `
  }), { info: {} })
