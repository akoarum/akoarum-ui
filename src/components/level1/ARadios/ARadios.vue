<template>
  <ul class="aRadios">
    <template v-for="item in items">
      <li :key="item.id" class="aRadios__item">
        <label class="aRadios__label">
          <input
            v-model="valueModel"
            v-bind="$attrs"
            :name="name"
            :value="item.value"
            type="radio"
            class="aRadios__input"
          />
          <span class="aRadios__text">{{ item.label }}</span>
        </label>
      </li>
    </template>
  </ul>
</template>

<script lang="ts">
import { defineComponent, computed, SetupContext } from '@vue/composition-api'
import { FormOption } from '~/types/FormOption'

interface Props {
  name: string | number
  items: FormOption[]
  value: string
}

export default defineComponent({
  inheritAttrs: false,
  model: {
    event: 'change'
  },
  props: {
    name: { type: String, required: true },
    items: { type: Array, required: true },
    value: { type: [String, Number], required: true }
  },
  setup(props: Props, context: SetupContext) {
    const valueModel = computed({
      get: (): string => props.value,
      set: (newValue: string) => {
        context.emit('change', newValue)
      }
    })

    return { valueModel }
  }
})
</script>

<style lang="scss" scoped>
.aRadios__item {
  font-size: 15px;
}

.aRadios__item + .aRadios__item {
  margin-top: 8px;
}

.aRadios__label {
  position: relative;
  display: inline-block;
  padding-left: 24px;
}

.aRadios__input {
  position: absolute;
  left: 0;
  top: 0;
  width: 4px;
  height: 4px;
  opacity: 0;

  &:checked + .aRadios__text::after {
    opacity: 1;
  }
  &[aria-invalid] + .aRadios__text {
    color: $color-caution;

    &::before {
      border-color: $color-caution;
      background: #fff7f7;
    }

    &::after {
      background: $color-caution;
    }
  }
}

.aRadios__text {
  &::before,
  &::after {
    position: absolute;
    content: '';
    border-radius: 50%;
  }

  &::before {
    top: 2px;
    left: 0;
    border: 2px solid $color-base;
    width: 14px;
    height: 14px;
    background: #fff;
  }

  &::after {
    top: 7px;
    left: 5px;
    width: 8px;
    height: 8px;
    background: $color-primary;
    opacity: 0;
    transition: opacity $duration $easing;
  }
}
</style>
