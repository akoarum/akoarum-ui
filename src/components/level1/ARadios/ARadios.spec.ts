import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import ARadios from './ARadios.vue'

describe('ARadios', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(ARadios, {
      propsData: {
        name: 'radio',
        value: '',
        items: [
          { id: 0, label: '項目1', value: 1 },
          { id: 1, label: '項目2', value: 2 },
          { id: 2, label: '項目3', value: 3 }
        ]
      },
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
