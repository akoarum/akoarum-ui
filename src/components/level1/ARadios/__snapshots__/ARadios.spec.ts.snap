// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ARadios initialized snapshot 1`] = `
<ul
  class="aRadios"
>
  <li
    class="aRadios__item"
  >
    <label
      class="aRadios__label"
    >
      <input
        class="aRadios__input"
        name="radio"
        type="radio"
        value="1"
      />
       
      <span
        class="aRadios__text"
      >
        項目1
      </span>
    </label>
  </li>
  <li
    class="aRadios__item"
  >
    <label
      class="aRadios__label"
    >
      <input
        class="aRadios__input"
        name="radio"
        type="radio"
        value="2"
      />
       
      <span
        class="aRadios__text"
      >
        項目2
      </span>
    </label>
  </li>
  <li
    class="aRadios__item"
  >
    <label
      class="aRadios__label"
    >
      <input
        class="aRadios__input"
        name="radio"
        type="radio"
        value="3"
      />
       
      <span
        class="aRadios__text"
      >
        項目3
      </span>
    </label>
  </li>
</ul>
`;
